cmake/Modules/FindGio.cmake

```diff
@@ -0,0 +1,27 @@
+# - Try to find Gio
+# Once done this will define
+#
+#  GIO_FOUND - system has Gio
+#  GIO_INCLUDE_DIRS - the Gio include directory
+#  GIO_LIBRARIES - the libraries needed to use Gio
+#  GIO_DEFINITIONS - Compiler switches required for using Gio
+
+# Use pkg-config to get the directories and then use these values
+# in the find_path() and find_library() calls
+find_package(PkgConfig)
+pkg_check_modules(PC_GIO gio-2.0)
+
+set(GIO_DEFINITIONS ${PC_GIO_CFLAGS})
+
+find_path(GIO_INCLUDE_DIRS gio.h PATHS ${PC_GIO_INCLUDEDIR} ${PC_GIO_INCLUDE_DIRS} PATH_SUFFIXES glib-2.0/gio/)
```

Comment:
Internally finds should be the old style FOO_INCLUDE_DIR and FOO_LIBRARY, then the outputs created by find_package_handle_standard_args will produce the standard FOO_INCLUDE_DIRS and FOO_LIBRARIES... Cmake has a lot of weird traditions.

Reply:
https://cmake.org/cmake/help/latest/module/FindPackageHandleStandardArgs.html sort of tangentially mentioned here.

Reply:
err im wrong you just set them manually in the canonical example. https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#a-sample-find-module

Reply:
I see. Well, I usually look at how KDE folks use CMake, and it seems to be a common practice to pass `LIBRARIES` / `DEFINITIONS` / `DIRS` directly to `find_*()`. [Example](https://github.com/KDE/krfb/blob/master/cmake/modules/FindPipeWire.cmake).